tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -385,6 +386,60 @@ LogicalResult ConvertTFLRelu1Op::matchAndRewrite(
   return success();
 }
 
+LogicalResult ConvertTFLRelu0To1Op::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_relu0to1_op = cast<TFL::Relu0To1Op>(op);
+
+  ShapedType input_type = tfl_relu0to1_op.x().getType().cast<ShapedType>();
+  ShapedType output_type =
+      tfl_relu0to1_op.getResult().getType().cast<ShapedType>();
+  // Not a ranked tensor output
+  if (!input_type || !output_type) return failure();
```

Comment:
At this point you don't know if ranked or not, you'd have to query hasRank

Reply:
I copied this from Relu, Relu1, and Relu6. Do we really need to check hasRank or is this simply a comment ambiguity?

Reply:
I looked at where this is used and I cannot see any place where rank or shape is being queried. Additionally the above does cast and not dyn_cast so we would have asserted here rather than returned.

As is I'd just drop these two lines: the input op has ShapedType in all legal cases and not rank is required anywhere here. One could even add an unranked test cases to verify.